adapters/metax/metax_test.go

```diff
@@ -0,0 +1,214 @@
+	}
+	b3n := assignBannerSize(b3)
+	assert.Equal(t, b3n.W, ptrutil.ToPtr(int64(336)))
+	assert.Equal(t, b3n.H, ptrutil.ToPtr(int64(280)))
+	assert.Same(t, b3, b3n)
+
+	b4 := &openrtb2.Banner{
+		Format: []openrtb2.Format{
+			{W: 300, H: 250},
+			{W: 728, H: 90},
+		},
+		W: ptrutil.ToPtr(int64(336)),
+	}
+	b4n := assignBannerSize(b4)
+	assert.Equal(t, b4n.W, ptrutil.ToPtr(int64(300)))
+	assert.Equal(t, b4n.H, ptrutil.ToPtr(int64(250)))
+	assert.NotSame(t, b4, b4n)
+
+	b5 := &openrtb2.Banner{}
+	b5n := assignBannerSize(b5)
+	assert.Nil(t, b5n.W)
+	assert.Nil(t, b5n.H)
+	assert.Same(t, b5, b5n)
+}
+
+func TestGetBidType(t *testing.T) {
+	tests := []struct {
+		bid     *openrtb2.Bid
+		bidtype openrtb_ext.BidType
+	}{
+		{&openrtb2.Bid{AdM: "", MType: openrtb2.MarkupBanner}, openrtb_ext.BidTypeBanner},
+		{&openrtb2.Bid{AdM: "", MType: openrtb2.MarkupVideo}, openrtb_ext.BidTypeVideo},
+		{&openrtb2.Bid{AdM: "", MType: openrtb2.MarkupNative}, openrtb_ext.BidTypeNative},
+		{&openrtb2.Bid{AdM: "", MType: openrtb2.MarkupAudio}, openrtb_ext.BidTypeAudio},
+		{&openrtb2.Bid{AdM: "", MType: 0}, ""},
+	}
+
+	for _, test := range tests {
+		bidType, err := getBidType(test.bid)
+		assert.Equal(t, test.bidtype, bidType)
+		if bidType == "" {
+			assert.NotNil(t, err)
+		}
+	}
+}
+
+func TestBuilder(t *testing.T) {
+	serverCfg := config.Server{}
+
+	cfg1 := config.Adapter{Endpoint: "https://hb.metaxads.com/prebid"}
+	builder1, err1 := Builder("test", cfg1, serverCfg)
+	assert.NotNil(t, builder1)
+	assert.Nil(t, err1)
+
+	// empty endpoint
+	cfg2 := config.Adapter{Endpoint: ""}
+	builder2, err2 := Builder("test2", cfg2, serverCfg)
+	assert.Nil(t, builder2)
+	assert.NotNil(t, err2)
+
+	// invalid endpoint
+	cfg3 := config.Adapter{Endpoint: "https://hb.metaxads.com/prebid?a={{}}"}
+	builder3, err3 := Builder("test3", cfg3, serverCfg)
+	assert.Nil(t, builder3)
+	assert.NotNil(t, err3)
+}
+
+func TestMakeRequests(t *testing.T) {
+	builder1, _ := Builder("metax", config.Adapter{Endpoint: "https://hb.metaxads.com/prebid?sid={{.PublisherId}}"}, config.Server{})
+	reqDatas1, err1 := builder1.MakeRequests(&openrtb2.BidRequest{
+		Imp: []openrtb2.Imp{
+			{
+				Ext: []byte(`
+					{
+						"bidder": {
+							"publisherId": 100,
+							"adunit": 2
+						}
+					}
+				`),
+			},
+		},
+		Ext: []byte(`{invalid json}`),
+	}, &adapters.ExtraRequestInfo{})
+	assert.Equal(t, 0, len(reqDatas1))
+	assert.Equal(t, 1, len(err1))
+
+	builder2, _ := Builder(
+		"metax",
+		config.Adapter{Endpoint: "https://hb.metaxads.com/prebid?sid={{.PublisherID}}&adunit={{.AdUnit}}&source=prebid-server"},
+		config.Server{},
+	)
+	reqDatas2, err2 := builder2.MakeRequests(&openrtb2.BidRequest{
+		Imp: []openrtb2.Imp{
+			{
+				Ext: []byte(`
+					{
+						"bidder": {
+							"publisherId": 100,
+							"adunit": 2
+						}
+					}
+				`),
+			},
+		},
+		Ext: []byte(`{invalid json}`),
+	}, &adapters.ExtraRequestInfo{})
+	assert.Equal(t, 0, len(reqDatas2))
+	assert.Equal(t, 1, len(err2))
+}
```

Comment:
tests should not look like this. 
Please rechange this for 2 files:
-metax_test.go where You test jsons samples
-params_test.go where You Test Valid and InvalidParams